observation/store.go

```diff
@@ -53,76 +69,97 @@ type Result struct {
 
 // SaveAll the observations against the provided dimension options and instanceID.
 func (store *Store) SaveAll(observations []*Observation) ([]*Result, error) {
-
 	results := make([]*Result, 0)
 
 	// handle the inserts separately for each instance in the batch.
 	instanceObservations := mapObservationsToInstances(observations)
 
-	for instanceID := range instanceObservations {
+	conn, err := store.pool.OpenPool()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := conn.Close(); err != nil {
+			log.Error(err, nil)
+		}
+	}()
 
-		query := buildInsertObservationQuery(instanceID, instanceObservations[instanceID])
+	for instanceID := range instanceObservations {
```

Comment:
if we captured the value as well as the key here, we could reference the value var instead of instanceObservations[instanceID]